index.bs

```diff
@@ -806,6 +811,34 @@ series of steps |steps|, run these steps:
 
 </div>
 
+## Session termination not initiated by the client ## {#web-transport-termination}
+
+<div algorithm="termination-initiated-by-server">
+Whenever a [=WebTransport session=] which is associated with a {{WebTransport}} |transport| is
+[=session/terminated=], run these steps:
+
+1. If |transport|'s [=[[State]]=] is `"closed"` or `"failed"`, abort these steps.
+1. Let |cleanly| be a boolean representing whether the HTTP/3 stream associated with the CONNECT
+   request that initiated |transport|'s [=[[Session]]=] is in the "Data Recvd" state. [[!QUIC]]
+1. [=Queue a network task=] with |transport| to run these steps:
+  1. If |transport|'s [=[[State]]=] is `"closed"` or `"failed"`, abort these steps.
+  1. Let |error| be TBD.
+  1. [=Cleanup=] |transport| with |error|, |error| and the negation of |cleanly|.
+
+</div>
+
+<div algorithm="termination-caused-by-connection-error">
```

Comment:
@vasilvv  Are there also transient connection errors that are recoverable that this shouldn't apply to?

Reply:
I don't believe there are any that we propagate to the renderer (there are some cases which we handle as a recoverable error, such as QUIC version mismatch, but those are handled internally by the network stack).  If we ever have recoverable errors, we'd probably want a different API for those.

Reply:
The current text is 

> Whenever a  connection  associated with a  WebTransport  transport  gets a connection error, run these steps

This means the session establishment has already been done. Failures in session establishment is covered by a step in https://w3c.github.io/webtransport/#initialize-webtransport-over-http. Does this help?